jsonrpc/filter_manager_test.go

```diff
@@ -11,6 +13,123 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func Test_GetLogsForQuery(t *testing.T) {
+	t.Parallel()
+
+	blockHash := types.StringToHash("1")
+
+	// Topics we're searching for
+	topic1 := types.StringToHash("4")
+	topic2 := types.StringToHash("5")
+	topic3 := types.StringToHash("6")
+
+	var topics = [][]types.Hash{{topic1}, {topic2}, {topic3}}
+
+	testTable := []struct {
+		name           string
+		query          *LogQuery
+		shouldFail     bool
+		expectedLength int
+	}{
+		{"Found matching logs, fromBlock < toBlock",
+			&LogQuery{
+				fromBlock: 1,
+				toBlock:   3,
+				Topics:    topics,
+			},
+			false, 3},
+		{"Found matching logs, fromBlock == toBlock",
+			&LogQuery{
+				fromBlock: 2,
+				toBlock:   2,
+				Topics:    topics,
+			},
+			false, 1},
+		{"Found matching logs, BlockHash present",
+			&LogQuery{
+				BlockHash: &blockHash,
+				Topics:    topics,
+			},
+			false, 1},
+		{"No logs found", &LogQuery{
+			fromBlock: 4,
+			toBlock:   5,
+			Topics:    topics,
+		}, false, 0},
+		{"Invalid block range", &LogQuery{
+			fromBlock: 10,
+			toBlock:   5,
+			Topics:    topics,
+		}, true, 0},
+	}
+
+	// setup test
+	store := &mockBlockStore{}
+	store.topics = []types.Hash{topic1, topic2, topic3}
+	store.setupLogs()
+
+	for i := 0; i < 5; i++ {
+		store.add(&types.Block{
+			Header: &types.Header{
+				Number: uint64(i),
+				Hash:   types.StringToHash(strconv.Itoa(i)),
+			},
+			Transactions: []*types.Transaction{
+				{
+					Value: big.NewInt(10),
+				},
+				{
+					Value: big.NewInt(11),
+				},
+				{
+					Value: big.NewInt(12),
+				},
+			},
+		})
+	}
```

Comment:
This can be a helper method

Reply:
What exactly? Adding multiple elements into the store with default values? Some test may require additional fields in the Header and Transaction object.

Reply:
Have a helper method like `appendBlocksToStore`, where you would add blocks (pass in an array of blocks)

Reply:
Here is the fix e44499d304fa50924cb8793f3c07d73e6b581794. If you don't think this is the way, say and we will improve it